src/EVM/SMT.hs

```diff
@@ -975,17 +1003,17 @@ getStore getVal sreads = do
         Nothing -> Map.insert addr (Map.singleton slot (fun addr slot)) store
 
 
-    queryValue :: Expr EWord -> IO W256
-    queryValue (Lit w) = pure w
-    queryValue w = do
-      let expr = toLazyText $ exprToSMT w
-      raw <- getVal expr
-      case parseCommentFreeFileMsg getValueRes (T.toStrict raw) of
-        Right (ResSpecific (valParsed :| [])) ->
-          case valParsed of
-            (_, TermSpecConstant sc) -> pure $ parseW256 sc
-            _ -> error "Internal Error: cannot parse model for storage index"
-        r -> parseErr r
+queryValue :: (Text -> IO Text) -> Expr EWord -> IO W256
+queryValue _ (Lit w) = pure w
+queryValue getVal w = do
+  let expr = toLazyText $ exprToSMT w
+  raw <- getVal expr
+  case parseCommentFreeFileMsg getValueRes (T.toStrict raw) of
+    Right (ResSpecific (valParsed :| [])) ->
+      case valParsed of
+        (_, TermSpecConstant sc) -> pure $ parseW256 sc
+        _ -> error "Internal Error: cannot parse model for storage index"
```

Comment:
We could use a more generic error here, since now the function is not only used for storage indices